spikeinterface/toolkit/postprocessing/unit_localization.py

```diff
@@ -68,13 +70,34 @@ def make_initial_guess_and_bounds(wf_ptp, local_contact_locations, max_distance_
 
     # bounds depend on initial guess
     bounds = ([x0[0] - max_distance_um, x0[1] - max_distance_um, 1, 0],
-              [x0[0] + max_distance_um,  x0[1] + max_distance_um, max_distance_um*10, max_alpha])
+              [x0[0] + max_distance_um, x0[1] + max_distance_um, max_distance_um * 10, max_alpha])
 
     return x0, bounds
 
 
-# ---- 
-# optimizer "least_square"
+def solve_monopolar_triangulation(wf_ptp, local_contact_locations, max_distance_um, optimizer):
+    x0, bounds = make_initial_guess_and_bounds(wf_ptp, local_contact_locations, max_distance_um)
+
+    if optimizer == 'least_square':
+        args = (wf_ptp, local_contact_locations)
+        output = scipy.optimize.least_squares(estimate_distance_error, x0=x0, bounds=bounds, args=args)
+        return tuple(output['x'])
+
+    if optimizer == 'minimize_with_log_penality':
+        x0 = x0[:3]
+        bounds = [(bounds[0][0], bounds[1][0]), (bounds[0][1], bounds[1][1]), (bounds[0][2], bounds[1][2])]
+        maxptp = wf_ptp.max()
+        args = (wf_ptp, local_contact_locations, maxptp)
+        output = scipy.optimize.minimize(estimate_distance_error_with_log, x0=x0, bounds=bounds, args=args)
+        # final alpha
+        q = ptp_at(*output['x'], 1.0, local_contact_locations)
```

Comment:
This 'x' direction should be controlable  by an argument in case we want to estimate motion in an another direction.
No ?

Reply:
Ah, it's not `x` as in the horizontal direction -- `output["x"]` is the localization result. It's accessing the final result from `minimize`, which returns a [`OptimizeResult`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.OptimizeResult.html#scipy.optimize.OptimizeResult) object that contains the optimization result in the field "x"

Reply:
oups sorry. I forgot this.